### ISSUE TITLE
[BUG] [TEST] BackgroundFetch unit tests fail - data race

### ISSUE THREAD

**Issue by turan18** [awslabs/soci-snapshotter#284]
**Describe the bug**

It appears there is a data race within the [TestBackgroundFetcherRun()](https://github.com/awslabs/soci-snapshotter/blob/106fdbb1ea903aa2535cb0395f8764d72892e796/fs/backgroundfetcher/background_fetcher_test.go#L70) test. Both test cases seem to fail ( background fetch all data for multiple and single span manager(s) )

**Steps To Reproduce**

`go clean -testcache`
`make test`

**Expected behavior**

All test cases should pass.

**Configuration (please complete the following information):**
 - OS: Amazon Linux 2
 - Snapshotter Version: Tip of Main
 - Containerd Version: containerd github.com/containerd/containerd v1.6.10 770bd0108c32f3fb5c73ae1264f7e503fe7b2661

**Additional context**

Error

```
WARNING: DATA RACE
Read at 0x00c0003361d0 by goroutine 19:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun.func1()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:134 +0x38b
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1486 +0x47

Previous write at 0x00c0003361d0 by goroutine 41:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*countingWriter).Write()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:180 +0xb7
  github.com/awslabs/soci-snapshotter/fs/span-manager.(*SpanManager).addSpanToCache()
      /home/turyasin/github/soci-snapshotter/fs/span-manager/span_manager.go:362 +0xc6
  github.com/awslabs/soci-snapshotter/fs/span-manager.(*SpanManager).FetchSingleSpan()
      /home/turyasin/github/soci-snapshotter/fs/span-manager/span_manager.go:210 +0x3c8
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*sequentialLayerResolver).Resolve()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/resolver.go:85 +0x2c7
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*BackgroundFetcher).Run.func1()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher.go:156 +0x7e

Goroutine 19 (running) created at:
  testing.(*T).Run()
      /usr/lib/golang/src/testing/testing.go:1486 +0x724
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:100 +0x3ef
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1486 +0x47

Goroutine 41 (finished) created at:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*BackgroundFetcher).Run()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher.go:155 +0x284
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun.func1.2()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:124 +0x58
==================
==================
WARNING: DATA RACE
Read at 0x00c0003361d8 by goroutine 19:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun.func1()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:141 +0x4c4
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1486 +0x47

Previous write at 0x00c0003361d8 by goroutine 41:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*countingWriter).Write()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:179 +0x71
  github.com/awslabs/soci-snapshotter/fs/span-manager.(*SpanManager).addSpanToCache()
      /home/turyasin/github/soci-snapshotter/fs/span-manager/span_manager.go:362 +0xc6
  github.com/awslabs/soci-snapshotter/fs/span-manager.(*SpanManager).FetchSingleSpan()
      /home/turyasin/github/soci-snapshotter/fs/span-manager/span_manager.go:210 +0x3c8
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*sequentialLayerResolver).Resolve()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/resolver.go:85 +0x2c7
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*BackgroundFetcher).Run.func1()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher.go:156 +0x7e

Goroutine 19 (running) created at:
  testing.(*T).Run()
      /usr/lib/golang/src/testing/testing.go:1486 +0x724
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:100 +0x3ef
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1486 +0x47

Goroutine 41 (finished) created at:
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.(*BackgroundFetcher).Run()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher.go:155 +0x284
  github.com/awslabs/soci-snapshotter/fs/backgroundfetcher.TestBackgroundFetcherRun.func1.2()
      /home/turyasin/github/soci-snapshotter/fs/backgroundfetcher/background_fetcher_test.go:124 +0x58
==================
--- FAIL: TestBackgroundFetcherRun (22.73s)
    --- FAIL: TestBackgroundFetcherRun/background_fetcher_fetches_all_data_for_multiple_span_managers (22.54s)
        testing.go:1312: race detected during execution of test
    testing.go:1312: race detected during execution of test
FAIL
FAIL	github.com/awslabs/soci-snapshotter/fs/backgroundfetcher	30.156s

```

**Comment by Kern--** [comment]
What version of go are you using?

**Comment by turan18** [comment]
> What version of go are you using?

go version go1.18.8 linux/amd64

**Comment by turan18** [comment]
Issues seems to be resolved after updating to the latest stable go version (1.19.4)

**Comment by djdongjin** [comment]
@Kern-- @turan18 if go1.19 can resolve the issue, should we update the go version we support?

https://github.com/awslabs/soci-snapshotter/blob/ae7e16948759bf449d8f7f8391264c2cedd3f9b9/go.mod#LL3C7-L3C7

https://github.com/awslabs/soci-snapshotter/blob/ae7e16948759bf449d8f7f8391264c2cedd3f9b9/cmd/go.mod#L3

**Comment by turan18** [comment]
> @Kern-- @turan18 if go1.19 can resolve the issue, should we update the go version we support?
> 
> https://github.com/awslabs/soci-snapshotter/blob/ae7e16948759bf449d8f7f8391264c2cedd3f9b9/go.mod#LL3C7-L3C7
> 
> https://github.com/awslabs/soci-snapshotter/blob/ae7e16948759bf449d8f7f8391264c2cedd3f9b9/cmd/go.mod#L3

Reopening for visibility/discussion

**Comment by Kern--** [comment]
@rdpsin you had an idea for how we can fix this while supporting go 1.18, right?
